packages/flutter_tools/lib/src/runner/flutter_command_runner.dart

```diff
@@ -300,7 +303,12 @@ class FlutterCommandRunner extends CommandRunner<void> {
     }
 
     _machineFlagPresentInAnyCliArg = args.contains('--${FlutterGlobalOptions.kMachineFlag}');
-    return super.run(args);
+    return super.run(args).then((_) {
+      if (exitWithCodeOne) {
+        // No need to print anything because the help was already printed.
+        throwToolExit('', exitCode: 1);
```

Comment:
Can we just call `exitWithHooks(1, shutdownHooks: ...)` instead? I'm pretty sure this will result in `Error: ` being printed.

Reply:
Done :+1: